examples/dmd/wave_equation.cpp

```diff
@@ -0,0 +1,621 @@
+
+            if (visualization)
+            {
+               sout << "solution\n" << *mesh << u_gf << flush;
+            }
+
+            if (visit)
+            {
+               visit_dc.SetCycle(ti);
+               visit_dc.SetTime(t);
+               visit_dc.Save();
+            }
+        }
+       oper.SetParameters(u);
+
+   }
+
+   // 9. Save the final solution. This output can be viewed later using GLVis:
+   //    "glvis -m wave_equation.mesh -g wave_equation-final.gf".
+   {
+      ofstream osol("wave_equation-final.gf");
+      osol.precision(precision);
+      u_gf.Save(osol);
+      dudt_gf.Save(osol);
+   }
+
+   // 10. Calculate the DMD modes.
+    if (myid == 0 && rdim != -1 && ef != -1)
+    {
+        std::cout << "Both rdim and ef are set. ef will be ignored." << std::endl;
+    }
+
+
+   Vector true_solution_u(u_gf.GetTrueVector().Size());
+   
+
+   true_solution_u = u_gf.GetTrueVector();
+
+
+   dmd_prediction_timer.Start();
+
+   // 11. Predict the state at t_final using DMD.
+   if (myid == 0)
+   {
+      std::cout << "Predicting temperature using DMD" << std::endl;
+   }
+
+   curr_window = 0;
+   
+   CAROM::Vector* result_u = dmd_u[curr_window]->predict(ts[0]);
+   Vector initial_dmd_solution_u(result_u->getData(), result_u->dim());
+   u_gf.SetFromTrueDofs(initial_dmd_solution_u);
+
+   VisItDataCollection dmd_visit_dc("DMD_Wave_Equation", mesh);
+   dmd_visit_dc.RegisterField("solution", &u_gf);
+   if (visit)
+   {
+      dmd_visit_dc.SetCycle(0);
+      dmd_visit_dc.SetTime(0.0);
+      dmd_visit_dc.Save();
+   }
+
+   delete result_u;
+
+   for (int i = 1; i < ts.size(); i++)
+   {
+      if (i == ts.size() - 1 || (i % vis_steps) == 0)
+       {
+
+         if(visit)
+         {
+            result_u = dmd_u[curr_window]->predict(ts[i]);
+            Vector dmd_solution_u(result_u->getData(), result_u->dim());
+            u_gf.SetFromTrueDofs(dmd_solution_u);
+            dmd_visit_dc.SetCycle(i);
+            dmd_visit_dc.SetTime(ts[i]);
+            dmd_visit_dc.Save();
+            delete result_u;
+         }
+
+         if (i % windowNumSamples == 0 && i < ts.size()-1) 
+         {
+            delete dmd_u[curr_window];
+            curr_window++;
+         }
+      }
+
+   }
+
+   dmd_prediction_timer.Stop();
+
+   result_u = dmd_u[curr_window]->predict(t_final);
+
+   // 11. Calculate the relative error between the DMD final solution and the true solution.
+  
+   Vector dmd_solution_u(result_u->getData(), result_u->dim());
+   Vector diff_u(true_solution_u.Size());
+   subtract(dmd_solution_u, true_solution_u, diff_u);
+   double tot_diff_norm_u = sqrt(InnerProduct(MPI_COMM_WORLD, diff_u, diff_u));
+   double tot_true_solution_u_norm = sqrt(InnerProduct(MPI_COMM_WORLD,
+                                           true_solution_u, true_solution_u));
+
+                                          
+   cout << "myid = " << myid << endl;
+   cout << "tot_diff_norm_u = " << tot_diff_norm_u << endl;
```

Comment:
Output should only be on `myid == 0`.

Reply:
`myid` has been removed from this(serial time-windowing DMD) implementation.  Should I later add parallelism to this example, I will keep this in mind.